### PR TITLE
Update release checklist from 2.1.520, 3.1.111 & 5.0.102 releases

### PR DESCRIPTION
**Pull request by dseefeld**

### CODE REVIEW

**Review by dagood — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/releases/release_checklist.md [hunk 1]
```diff
@@ -109,29 +113,31 @@
...
-1.  - [ ] [Internal] If a build hits `fatal: reference is not a tree: cb5f173b`, [make the commit available on a fork and switch to it temporarily](https://github.com/dotnet/source-build/tree/release/3.1/Documentation/servicing/make-commit-available.md).
+1.  - [ ] [Internal] If a build hits `fatal: reference is not a tree: cb5f173b`, [make the commit available on a fork and switch to it temporarily](https://github.com/dotnet/source-build/tree/release/master/Documentation/servicing/make-commit-available.md).
 1.  - [ ] [Internal] Add source-build team as reviewers.
 1.  - [ ] [Internal] When CI is green and two reviewers approve, merge.
       - Avoid squash/rebase: nice to preserve commit hashes. However, there are no known dependencies on *source-build* commits being preserved.
-1.  - [ ] Create and push tags on the post-merge commit. [/Documentation/servicing/tagging.md](https://github.com/dotnet/source-build/tree/release/3.1/Documentation/servicing/tagging.md)
+1.  - [ ] Create and push tags on the post-merge commit. [/Documentation/servicing/tagging.md](https://github.com/dotnet/source-build/tree/release/master/Documentation/servicing/tagging.md)
```
Comment (crummel) [inline .github/ISSUE_TEMPLATE/releases/release_checklist.md:129]: Could you add the bit about waiting for the other repos' merges to go through before tagging?  Something like
```suggestion
1.  - [ ] After all contributing repos have fixed any missing merges (from the `fatal: reference is not a tree` item above), Make a new PR to move any repos changed back to their official source.
1. - [ ] Create and push tags on the post-merge commit. [/Documentation/servicing/tagging.md](https://github.com/dotnet/source-build/tree/release/master/Documentation/servicing/tagging.md)
```

**Review by crummel — APPROVED** [review]
Just one small doc fix that was assigned to me if you don't mind including it in this batch  :)

File: .github/ISSUE_TEMPLATE/releases/release_checklist.md (hunk 1 above)
Comment (dseefeld) [reply .github/ISSUE_TEMPLATE/releases/release_checklist.md]: Thanks.  Added.

File: .github/ISSUE_TEMPLATE/releases/release_checklist.md [hunk 2]
```diff
@@ -109,29 +113,31 @@
...
-1.  - [ ] [Internal] [3.1] In a dev branch on your GitHub fork of source-build, clean up.
-      - [ ] [Internal] [3.1] Convert internal URIs to public in `eng/Version.Details.xml`.
+1.  - [ ] [Internal] [3.1 / 5.0] In a dev branch on your GitHub fork of source-build, clean up.
+      - [ ] [Internal] [3.1 / 5.0] Convert internal URIs to public in `eng/Version.Details.xml`.
         - `https://dev.azure.com/dnceng/internal/_git/<org>-<repo>` => `https://github.com/<org>/<repo>`
-      - [ ] [Internal] [3.1] Remove internal package feeds from source-build.
+      - [ ] [Internal] [3.1 / 5.0] Remove internal package feeds from source-build.
         - `/NuGet.Config`
         - `/smoke-testNuGet.Config`
+        - repo projects - Update  `<SourceDirectory>dotnet-XXXXX</SourceDirectory>` to `<SourceDirectory>XXXXX</SourceDirectory>`.  If the `SourceDirectory` is the same as the RepositoryName, the element can be removed.  Issue to track consistent repo names: [#1608](https://github.com/dotnet/source-build/issues/1608)
```
[the remark this answers is not in the document]
Comment (dseefeld) [reply .github/ISSUE_TEMPLATE/releases/release_checklist.md]: Yes.  Makes more sense.

**Pull request merged**